ironfish/src/rpc/routes/wallet/rescan.ts

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { ApiNamespace } from '../namespaces'
+import { routes } from '../router'
+import { AssertHasRpcContext } from '../rpcContext'
+
+export type RescanRequest = {
+  follow?: boolean
+  from?: number
+}
+export type RescanResponse = { sequence: number; startedAt: number; endSequence: number }
+
+export const RescanRequestSchema: yup.ObjectSchema<RescanRequest> = yup
+  .object({
+    follow: yup.boolean().optional(),
+    from: yup.number().optional(),
+    full: yup.boolean().optional(),
+  })
+  .defined()
+
+export const RescanResponseSchema: yup.ObjectSchema<RescanResponse> = yup
+  .object({
+    sequence: yup.number().defined(),
+    endSequence: yup.number().defined(),
+    startedAt: yup.number().defined(),
+  })
+  .defined()
+
+routes.register<typeof RescanRequestSchema, RescanResponse>(
+  `${ApiNamespace.wallet}/rescan`,
+  RescanRequestSchema,
+  async (request, context): Promise<void> => {
+    AssertHasRpcContext(request, context, 'wallet', 'logger')
+
+    while (context.wallet.scanner.running) {
+      context.logger.warn('Aborting scanning to start a full wallet rescan.')
+      await context.wallet.scanner.abort()
+    }
+
+    await context.wallet.resetAccounts({
+      resetScanningEnabled: true,
```

Comment:
I'm not sure if we want to reset `scanningEnabled` here by default, maybe a param for it? At least if we do, we should document that this will resume scanning for all accounts with scanning off.

That said, I feel like this RPC route is near deprecation anyway, since I feel like usually users will only want to rescan one particular account, and you can effectively do that with `wallet/resetAccount`

Reply:
I was just keeping backwards compatability with the previous code. I can remove this. I think it should not be reset personally.

Reply:
I'd lean toward removing it. I don't see where it reset `scanningEnabled` in the previous code, but maybe I'm missing it

Reply:
Done